work_queue/src/work_queue_worker.c

```diff
@@ -2092,28 +2090,43 @@ workspace_prepare is called every time we connect to a new manager,
 static int workspace_prepare()
 {
 	debug(D_WQ,"preparing workspace %s",workspace);
+
 	char *cachedir = string_format("%s/cache",workspace);
 	int result = create_dir(cachedir,0777);
 	free(cachedir);
 
 	char *tmp_name = string_format("%s/cache/tmp", workspace);
```

Comment:
@btovar Do you have any idea why the temp directory was set up inside the cache directory?  Would it make sense to put it at the same level in the hierarchy?

Reply:
I don't. It puzzled me too when I saw it recently.